DSESwift/DataSyncer.swift

```diff
@@ -8,37 +8,398 @@
+        self.persistentPeriod = self.config.localPersistancePeriod!
+    }
+    
+    /**
+     * this will error check it's own configuration
+     */
+    func setConfig(config: DSEConfig) throws -> (configChanged: Bool, syncInterval: Double, localPersistancePeriod: Double) {
+        var configChanged = false
+        if let syncInterval = config.syncInterval {
+            if (syncInterval < 0) {throw DataSyncerError.InvalidSyncRate}
+            configChanged = configChanged || self.syncRate != syncInterval
+            self.syncRate = syncInterval
+        }
+        
+        if let localPersistancePeriod = config.localPersistancePeriod {
+            if (localPersistancePeriod < 0) {throw DataSyncerError.InvalidPersistentPeriod}
+            configChanged = configChanged || self.persistentPeriod != localPersistancePeriod
+            self.persistentPeriod = localPersistancePeriod
+        }
+        
+        // return new values that will be used
+        return (configChanged, self.syncRate, self.persistentPeriod)
+    }
+    
+    func initialize()  {
+        dispatch_async(data_syncer_process_queue, {
+            do{
+                try self.downloadSensorProfiles()
+                print("DSE Initialization Completed")
+                self.initialized = true
+                self.delegate?.onInitializationCompleted()
+            }catch{
+                self.delegate?.onInitializationFailed(error)
+            }
+        })
+    }
+    
+    func enablePeriodicSync() {
+        dispatch_async(data_syncer_process_queue, {
+            if (self.initialized){
+                if self.timer == nil {
+                    self.timer = NSTimer.scheduledTimerWithTimeInterval(self.syncRate, target: self, selector: "sync", userInfo: nil, repeats: true);
+                }
+                self.timer!.fire()
+            }else{
+                print("DSE is not initialized")
+            }
+        })
+    }
+
+    func disablePeriodicSync(){
+        if self.timer == nil {
+            timer!.invalidate()
+        } else {
+            print("timer is nil")
+        }
+    }
+    
+    /**
+     * Synchronize data in local and remote storage.
+     * Is executed asynchronously.
+     */
+    public func sync() {
+        
+        dispatch_promise(on: data_syncer_process_queue, body:{
+            return Promise<Void> { fulfill, reject in
+                // step 1
+                try self.processDeletionRequests()
+                // step 2
+                try self.uploadSensorDataToRemote()
+                // step 3
+                try self.downloadSensorsFromRemote()
+                // step 4
+                try self.downloadSensorsDataFromRemote()
+                // step 5
+                try self.cleanLocalStorage()
+                fulfill()
+            }
+        }).then({
+            self.delegate?.onSyncCompleted()
+        }).error({ error in
+            print(error)
```

Comment:
do we remove this print or give it more context ie. "There has been an error in the syncing process: <errorcode>"